### ISSUE TITLE
Add Nextcloud and Gitea OAuth2 clients

### ISSUE THREAD

**Issue by xf-** [presentator/presentator#101]
I would like to add a generic oauth provider like nextcloud (self-hosted cloud) or gitea (self-hosted git). GitLab works already on oauth with hardcoded paths.

e.g.
https://docs.gitea.io/en-us/oauth2-provider/

**Comment by ganigeorgiev** [comment]
If you are asking for a truly generic/universal client configuration - that is kind of difficult to implement, since some clients may require additional response processing to unify the returned field keys and values. 
Presentator uses [yii2-authclient](https://github.com/yiisoft/yii2-authclient) extensions and by default currently only Facebook, Google, GitHub and GitLab are supported (for configuration options check https://github.com/ganigeorgiev/presentator/blob/master/common/config/params.php#L63)

I'll consider adding nextcloud and gitea clients support in v2 release.

**Comment by ganigeorgiev** [comment]
Merged in master and included in v2.1.0 release. Sample required configuration:
```
'gitea' => [
    'class'        => 'presentator\api\authclients\Gitea',
    'serviceUrl'   => 'https://my-gitea.com',
    'clientId'     => '',
    'clientSecret' => '',
],
```
